build/lib/release.sh

```diff
@@ -374,10 +374,14 @@ function kube::release::create_docker_images_for_server() {
         ln "${binary_file_path}" "${docker_build_path}/${binary_name}"
         ln "${KUBE_ROOT}/build/nsswitch.conf" "${docker_build_path}/nsswitch.conf"
         chmod 0644 "${docker_build_path}/nsswitch.conf"
-        cat <<EOF > "${docker_file_path}"
+        if [[ -f "${KUBE_ROOT}/build/server-image/${binary_name}/Dockerfile" ]]; then
+          ln "${KUBE_ROOT}/build/server-image/${binary_name}/Dockerfile" "${docker_file_path}"
+        else
+          cat <<EOF > "${docker_file_path}"
```

Comment:
TODO: we should really use a fixed dockerfile like the one above and docker arg, we haven't needed to use sed etc. for a very very long time.

I'll track this separately.

Reply:
Done in PR https://github.com/kubernetes/kubernetes/pull/99015.